src/sentry/integrations/github_enterprise/integration.py

```diff
@@ -0,0 +1,217 @@
+                "id": form_data.get('client-id'),
+                "secret": form_data.get('client-secret'),
+            })
+
+            return pipeline.next_step()
+
+        project_form = InstallationForm()
+
+        return render_to_response(
+            template='sentry/integrations/github-enterprise-config.html',
+            context={
+                'form': project_form,
+            },
+            request=request,
+        )
+
+
+class GitHubEnterpriseIntegrationProvider(GitHubIntegrationProvider):
+    key = 'github-enterprise'
+    name = 'GitHub Enterprise'
+    metadata = metadata
+
+    def get_pipeline_views(self):
+        identity_pipeline_config = {
+            'oauth_scopes': (),
+            'redirect_url': absolute_uri('/extensions/github-enterprise/setup/'),
+        }
+
+        identity_pipeline_view = NestedPipelineView(
+            bind_key='identity',
+            provider_key='github-enterprise',
+            pipeline_cls=IdentityProviderPipeline,
+            config=identity_pipeline_config,
+        )
+
+        return [InstallationConfigView(),
+                GitHubEnterpriseInstallationRedirect(),
+                identity_pipeline_view]
+
+    def get_installation_info(self, url, access_token, installation_id):
+        session = http.build_session()
+        resp = session.get(
+            # why does this 404?
+            # Not Found for url: https://35.232.149.196/api/v3/app/installations/27
+            # https://developer.github.com/enterprise/2.13/v3/apps/#get-a-single-installation
+            'https://{}/api/v3/app/installations/{}'.format(url, installation_id),
+            # 'https://{}/api/v3/app/installations/'.format(url, installation_id),
+            headers={
+                'Authorization': 'Bearer %s' % get_jwt(),
+                'Accept': 'application/vnd.github.machine-man-preview+json',
+            },
+            verify=False
+        )
+        resp.raise_for_status()
+        installation_resp = resp.json()
+
+        resp = session.get(
+            'https://{}/api/v3/user/installations'.format(url),
+            params={'access_token': access_token},
+            headers={'Accept': 'application/vnd.github.machine-man-preview+json'},
+            verify=False
+        )
+        resp.raise_for_status()
+        user_installations_resp = resp.json()
+
+        # verify that user actually has access to the installation
+        for installation in user_installations_resp['installations']:
+            if installation['id'] == installation_resp['id']:
+                return installation_resp
+
+        return None
+
+    def build_integration(self, state):
+        identity = state['identity']['data']
+        installation_data = state['installation_data']
+        # user = get_user_info(installation_data['url'], identity['access_token'])
+        # this doesn't work yet (404s):
+        installation = self.get_installation_info(
```

Comment:
this is the main issue right now.

Reply:
I think you were maybe using the wrong id?